packages/valory/skills/registration_abci/rounds.py

```diff
@@ -166,3 +165,19 @@ class AgentRegistrationAbciApp(AbciApp[Event]):
     event_to_timeout: Dict[Event, float] = {
         Event.ROUND_TIMEOUT: 30.0,
     }
+    db_pre_conditions: Dict[AppState, List[str]] = {
+        RegistrationStartupRound: [],
+        RegistrationRound: [],
+    }
+    db_post_conditions: Dict[AppState, List[str]] = {
+        FinishedRegistrationRound: [
+            get_name(BaseSynchronizedData.participants),
+            get_name(BaseSynchronizedData.all_participants),
+            get_name(BaseSynchronizedData.period_count),
+        ],
+        FinishedRegistrationFFWRound: [
+            get_name(BaseSynchronizedData.participants),
+            get_name(BaseSynchronizedData.all_participants),
+            get_name(BaseSynchronizedData.period_count),
```

Comment:
There's a question whether we should say `period_count` is a precondition instead. We could also have a set of preconditions that are just provided framework side.

Reply:
Strictly speaking `period_count` is not used in this app so I will remove it and provide it framework side. Otherwise this is a bit arbitrary.

Reply:
The three values `nb_participant`, `round_count` and `period_count` are special values provided framework side at all times.